### PR TITLE
TRUS-2669

### PR DESCRIPTION
**Pull request by jassalrichy**
Added Agents Details, Agent Address mapper
Added submit CTA to check answers
Added registration progress url to config
Added estates store connector

### CODE REVIEW

File: app/connector/EstateStoreConnector.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import config.FrontendAppConfig
+import javax.inject.Inject
+import models.mappers.AgentDetails
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class EstateStoreConnector @Inject()(http: HttpClient, config: FrontendAppConfig) {
```
Comment (adamconder) [inline app/connector/EstateStoreConnector.scala:27]: This is not required for the task list and the API does not exist in the estates-store.

File: app/controllers/CheckYourAnswersController.scala [hunk 2]
```diff
@@ -55,4 +65,22 @@ class CheckYourAnswersController @Inject()(
...
   }
+
+  def onSubmit() = actions.authWithData.async {
+    implicit request =>
+
+      agentMapper.build(request.userAnswers) match {
+        case Some(agentDetails) =>
+          // TODO call estates store service with Agent Details and return to Estate Hub service
+          Logger.info("[CheckYourAnswersController][submit] agent details generated for submit.")
+//          estateStoreConnector.setTaskComplete(agentDetails) map { _ =>
```
Comment (adamconder) [inline app/controllers/CheckYourAnswersController.scala:76]: Estates-store call is not required.

File: app/controllers/CheckYourAnswersController.scala [hunk 3]
```diff
@@ -55,4 +65,22 @@ class CheckYourAnswersController @Inject()(
 
       Ok(view(sections))
   }
+
+  def onSubmit() = actions.authWithData.async {
+    implicit request =>
+
+      agentMapper.build(request.userAnswers) match {
+        case Some(agentDetails) =>
+          // TODO call estates store service with Agent Details and return to Estate Hub service
```
Comment (adamconder) [inline app/controllers/CheckYourAnswersController.scala:74]: How are you getting on with sending the information `agentDetails` to the estates backend?

File: app/connector/EstateStoreConnector.scala (hunk 1 above)
Comment (jassalrichy) [reply app/connector/EstateStoreConnector.scala]: Removed

File: app/controllers/CheckYourAnswersController.scala (hunk 2 above)
Comment (jassalrichy) [reply app/controllers/CheckYourAnswersController.scala]: Removed

File: app/controllers/CheckYourAnswersController.scala (hunk 3 above)
Comment (jassalrichy) [reply app/controllers/CheckYourAnswersController.scala]: Updated

File: test/controllers/AgentTelephoneNumberControllerSpec.scala [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
 package controllers
 
-import base.SpecBase
+import base.{RegistrationSpecBase, SpecBase}
```
Comment (jaynabonne) [inline test/controllers/AgentTelephoneNumberControllerSpec.scala:19]: Is SpecBase still used in these specs now that you're using RegistrationSpecBase?

Comment (jassalrichy) [reply test/controllers/AgentTelephoneNumberControllerSpec.scala]: Updated

**Pull request merged**